common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -119,15 +204,27 @@ public static String getLocalIpAddress(int timeout) {
       // Make sure that the address is actually reachable since in some network configurations
       // it is possible for the InetAddress.getLocalHost() call to return a non-reachable
       // address e.g. a broadcast address
-      if (address.isLoopbackAddress() || !address.isReachable(timeout)) {
+      if (address.isAnyLocalAddress() || address.isLoopbackAddress()
+          || !address.isReachable(timeout) || !(address instanceof Inet4Address)) {
         Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+
+        // Make getNetworkInterfaces have the same order of network interfaces as listed on
+        // unix-like systems. This optimization can help avoid to get some special addresses, such
+        // as loopback address"127.0.0.1", virtual bridge address "192.168.122.1" as far as
```

Comment:
On unix-like systems (e.g. CentOS, Mac OS X), NetworkInterface.getNetworkInterfaces() gets a revered interface list as shown by using "ifconfig".  For example, the last interface on CentOS is usually "virbr0" having IP "inet addr:192.168.122.1", which is not a good selection for Tachyon.